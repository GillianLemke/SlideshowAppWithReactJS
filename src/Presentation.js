import React from 'react';
import './index.css';
import Slides from './slides'

export default class Presentations extends React.Component {
  constructor() {
    super();
    this.state = {
      title:'',
      listOfSlides:[],
    };
  }

  getTitleFromAPI() {
    return $.getJSON('google.com/pres')
      .then((data) => {
        this.setState({ title:data.results })
      });
  }

  // should the url be passed into this method?
  getListOfSlides() {
    let i = 0;
    return $.getJSON('google.com/slide/#')
      .then((data) => {
        // this condition needs to be redone
        for (i = 0; i < listOfSlides.length; i++) {
          // need to get individual portion of string
          this.setState({ listOfSlides:data.results })
        }
      });
  }

  render() {
    return (<Slides />);
  }
}
