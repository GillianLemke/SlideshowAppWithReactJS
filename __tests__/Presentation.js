import React from 'react'
import Presentation from '../src/Presentation.js'
import {shallow} from 'enzyme' // could also mount
import toJson from 'enzyme-to-json'

describe('Presentations', () => {
  it('renders', () => {
    const wrapper = shallow(<Presentation />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
  // does this test work?
  it('contains a title', () => {
    const wrapper = shallow(<Presentation title='pres1' />)
    expect(toJson(wrapper)).toMatchSnapshot()
  });
});
