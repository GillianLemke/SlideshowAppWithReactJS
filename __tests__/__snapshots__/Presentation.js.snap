exports[`Presentations contains a title 1`] = `<Slides />`;

exports[`Presentations renders 1`] = `<Slides />`;
